kong/plugins/jwt/handler.lua

```diff
@@ -118,8 +118,9 @@ local function do_authentication(conf)
   end
 
   local claims = jwt.claims
-
-  local jwt_secret_key = claims[conf.key_claim_name]
+  local header = jwt.header
+  
+  local jwt_secret_key = claims[conf.key_claim_name] or header[conf.key_claim_name]
```

Comment:
If the claim is found in both `payload` and `header`, the `header` one is never used. And I think you did this for better backward compatibility. I was just thinking, could we try both in that scenario and if one works (to signature verification) then continue. For that you may want to extract signature verification (and cache/db key loading) to its own function(s). And then just try both. If one works, then it is ok. @brycehemme Any thoughts on this (yes, it makes it a bit more complex, but also a bit more robust. Of course in most cases the `payload` or the `header` one is `nil` so you will still in most cases try signature verification just once?

Reply:
That makes sense although it does seem to add unnecessary complexity, in my opinion. It seems like a major edge case where the same claim would exist in the body and the header, each having a different value, and one of them being intended to be used for looking up the proper key for checking the signature. Regardless, if you think this is necessary I can certainly make the change.

Additionally, I had considered instead handling the naming of the value for the "config.key_claim_name" as explicitly referring to the header. So, something like this -

```
config.key_claim_name=header.kid
```
So, if the value is prefixed with "header." then it will explicitly look at the header claims and if it isn't prefixed with "header." then it will use the standard claims. 

Thoughts on this approach vs. what you suggested vs. the way I've already implemented it?

Also, thanks for kicking the build!

Reply:
@bungle - Can I get your thoughts on the above comments? I'm happy to address however you feel is appropriate. Thanks!

Reply:
I think we have a few options:

1. leave it as is
2. use prefix `header.` or something to refer claims in JWT headers
3. try both if first one didn't work
4. add new config value, e.g. `config.key_claim_in=payload|header` (default being `payload`)
5. similar to 4. but ordered array (of these two possible values) `config.key_claim_in=header,payload`
6. A boolean value like `config.key_claim_in_header=true|false (`false` being default)

Maybe there are other options as well. @brycehemme what do you think? I don't have a favourite.

Reply:
I would opt for leaving it as is. I implemented #2 above but after I completed it I didn't really like having the logic in the code. Although it would work just fine, it just makes things more confusing than necessary, in my opinion. As you can tell, I prefer reducing complexity unless it is absolutely necessary. Any issues with accepting this PR in this case?